examples/text/plot_hashing_vs_dict_vectorizer.py

```diff
@@ -3,109 +3,343 @@
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+print(f"Found {len(vectorizer.get_feature_names_out())} unique terms")
+
+# %%
+# We see that using the :func:`~sklearn.feature_extraction.text.CountVectorizer`
+# implementation is approximately twice as fast as using the
+# :func:`~sklearn.feature_extraction.DictVectorizer` along with the simple
+# function we defined for mapping the tokens. The reason is that
+# :func:`~sklearn.feature_extraction.text.CountVectorizer` produces a sparse
+# representation of the counts using `scipy.sparse.csr_matrix`.
+#
+# Now we make a similar experiment with the
+# :func:`~sklearn.feature_extraction.text.HashingVectorizer`, which is
+# equivalent to combining the “hashing trick” implemented by the
+# :func:`~sklearn.feature_extraction.FeatureHasher` class and the text
+# preprocessing and tokenization of the
+# :func:`~sklearn.feature_extraction.text.CountVectorizer`.
+
+from sklearn.feature_extraction.text import HashingVectorizer
+
+t0 = time()
+vectorizer = HashingVectorizer(n_features=2**18)
+vectorizer.fit_transform(raw_data)
+duration = time() - t0
+dict_count_vectorizers["vectorizer"].append(vectorizer.__class__.__name__)
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+
+# %%
+# TfidfVectorizer
+# ---------------
+#
+# In a large text corpus, some words appear with higher frequency (e.g. “the”,
+# “a”, “is” in English) which do not carry meaningful information about the
+# actual contents of a document. If we were to feed the word count data directly
+# to a classifier those very common terms would shadow the frequencies of rarer
+# yet more interesting terms. In order to re-weight the count features into
+# floating point values suitable for usage by a classifier it is very common to
+# use the tf–idf transform as implemented by the
+# :func:`~sklearn.feature_extraction.text.TfidfTransformer`. Tf stands for
+# "term-frequency" while "tf–idf" means term-frequency times inverse
+# document-frequency.
+#
+# We now benchmark the :func:`~sklearn.feature_extraction.text.TfidfVectorizer`,
+# which is equivalent to combining the tokenization and occurrence counting of
+# the :func:`~sklearn.feature_extraction.text.CountVectorizer` along with the
+# normalizing and weighting from a
+# :func:`~sklearn.feature_extraction.text.TfidfTransformer`.
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+t0 = time()
+vectorizer = TfidfVectorizer()
+vectorizer.fit_transform(raw_data)
+duration = time() - t0
+dict_count_vectorizers["vectorizer"].append(vectorizer.__class__.__name__)
+dict_count_vectorizers["speed"].append(data_size_mb / duration)
+print(f"done in {duration:.3f}s at {data_size_mb / duration:.3f}MB/s")
+print(f"Found {len(vectorizer.get_feature_names_out())} unique terms")
+
+fig, ax = plt.subplots(figsize=(12, 6))
```

Comment:
Let's split this large code cell to get the text output of the print statements right after them and the use the plot as conclusion:

```suggestion

# %%
# Let's conclude this notebook by summarizing all the recorded processing
# speeds in a single plot:
fig, ax = plt.subplots(figsize=(12, 6))
```